src/components/Tabs/Annotation/AnnotationTable.tsx

```diff
@@ -0,0 +1,118 @@
+import { CloseOutlined } from "@ant-design/icons";
+import { Table, TableProps } from "antd";
+import React, { ReactElement, useMemo } from "react";
+import styled from "styled-components";
+
+import { TagIconSVG } from "../../../assets";
+import { Dataset } from "../../../colorizer";
+import { FlexColumnAlignCenter, VisuallyHidden } from "../../../styles/utils";
+
+import IconButton from "../../IconButton";
+
+export type TableDataType = {
+  key: string;
+  id: number;
+  track: number;
+  time: number;
+};
+
+const StyledAntTable = styled(Table)`
+  .ant-table-row {
+    cursor: pointer;
+  }
+
+  &&&& .ant-table-cell {
+    padding: 4px 8px;
+  }
+`;
+
+type AnnotationTableProps = {
+  onClickObjectRow: (record: TableDataType) => void;
+  onClickDeleteObject: (record: TableDataType) => void;
+  dataset: Dataset | null;
+  ids: number[];
+};
+
+export default function AnnotationTable(props: AnnotationTableProps): ReactElement {
+  const tableColumns: TableProps<TableDataType>["columns"] = [
+    {
+      title: "Object ID",
+      dataIndex: "id",
+      key: "id",
+      sorter: (a, b) => a.id - b.id,
+    },
+    {
+      title: "Track ID",
+      dataIndex: "track",
+      key: "track",
+      sorter: (a, b) => a.track - b.track,
+    },
+    {
+      title: "Time",
+      dataIndex: "time",
+      key: "time",
+      sorter: (a, b) => a.time - b.time,
+    },
+    // Column that contains a remove button for the ID.
+    {
+      title: "",
+      key: "action",
+      render: (_, record) => (
+        <IconButton
+          type="text"
+          onClick={(event) => {
+            // Rows have their own behavior on click (jumping to a timestamp),
+            // so we need to stop event propagation so that the row click event
+            // doesn't fire.
+            event.stopPropagation();
+            props.onClickDeleteObject(record);
+          }}
+        >
+          <CloseOutlined />
+          <VisuallyHidden>
+            Remove ID {record.id} (track {record.track})
+          </VisuallyHidden>
+        </IconButton>
+      ),
+    },
+  ];
+
+  const tableData: TableDataType[] = useMemo(() => {
+    const dataset = props.dataset;
+    if (dataset) {
+      return props.ids.map((id) => {
```

Comment:
Suggestion: make this a method on dataset, so you can do `return dataset.getAnnotationTableData(props.ids)` in order to keep the data wrangling separate from the front end ui. look for other places to do the same type of thing.

Reply:
I started thinking about this one but it feels weird to have Dataset know about the specific use case for annotation tables? The data it's retrieving (track and time) are general enough that it seems okay to calculate with the existing API jere, but if this was doing anything more complex I would move it into a utilities file.